master/buildbot/schedulers/trysched.py

```diff
@@ -117,6 +120,9 @@ def activate(self):
 
     @defer.inlineCallbacks
     def deactivate(self):
+        if not self.enabled:
```

Comment:
base scheduler calls this method when a disable message is received, so the disable message will not actually disable.

Actually, this might work because self.enabled is set after calling the method.
..but the calling does not wait for results, and inlineCallback does not guarantee that the decorated method is called synchronously.

A unit test should test that to make sure it works for every version of twisted/python.